File: WikiPhytoScrapper/WikiPhytoScrapper/Models/PlantProperty.cs
using System.Collections.Generic;

namespace WikiPhytoScrapper.Models
{
    public class PlantProperty
    {
        public PlantProperty(PropertyCategory property)
        {
            this.PropertyCategory = property;
        }

        public string Name
        {
            get
            {
                return PropertyCategory.ToString();
            }
        }
        public List<string> Content { get; set; }

        private PropertyCategory PropertyCategory { get; set; }
    }

    public enum PropertyCategory
    {
        Name,
        Description,
        History,
        Dosis,
        Composition,
        HealthProperty,
        Indications,
        TherapeuticEffects,
        UndesirableEffects
    }
}

File: Website/src/models/plant-property.ts
export interface IPlantProperty {
    Name: string;
    Content: string[];
}

export class PlantProperty {
    name: string;
    content: string[];

    constructor(plantProperty: IPlantProperty | undefined) {
        if (!plantProperty) {
            this.name = '';
            this.content = [];
            return;
        }
        this.name = plantProperty.Name;
        this.content = plantProperty.Content;
    }
}
